libdisco/asymmetric_test.go

```diff
@@ -7,21 +7,68 @@ import (
 
 func TestSignVerify(t *testing.T) {
 	input := []byte("hi, how are you?")
-	kp := GenerateSigningKeyPair()
+	kp, err := GenerateSigningKeypair()
+
+	if err != nil {
+		t.Fatal("failed to generate a signing keypair")
+	}
 	sig := kp.Sign(input)
 
-	if kp.Verify(input, sig) != nil {
-		t.Fatal("Schnorr doesn't work")
+	if ok, err := kp.Verify(input, sig); !ok || err != nil {
+		t.Fatal("failed to verify signature with error : ", err)
 	}
 }
 
 func TestDeterministicSignatures(t *testing.T) {
-	kp := GenerateSigningKeyPair()
+	kp, err := GenerateSigningKeypair()
+	if err != nil {
+		t.Fatal("failed to generate a signing keypair")
+	}
 	input := []byte("hi, how are you?")
 	sig1 := kp.Sign(input)
 	sig2 := kp.Sign(input)
 
-	if !bytes.Equal(sig1, sig2) {
-		t.Fatal("Signatures are not deterministic doesn't work")
+	sig1Bytes := sig1.Encode()
+	sig2Bytes := sig2.Encode()
+
+	if !bytes.Equal(sig1Bytes[:], sig2Bytes[:]) {
+		t.Fatal("signatures are not deterministic doesn't work")
+	}
+}
+
+func BenchmarkSign(b *testing.B) {
+	input := []byte("benchmark how fast do I get signed")
+
+	for n := 0; n < b.N; n++ {
+		kp, err := GenerateSigningKeypair()
+
+		if err != nil {
+			b.Fatal("failed to generate signing keypair")
+		}
+		sig := kp.Sign(input)
+
+		if ok, err := kp.Verify(input, sig); !ok || err != nil {
+			b.Fatal("failed to verify signature with error : ", err)
```

Comment:
I don't think you're supposed to fail in a benchmark, since it's not a test per se : o